### PR TITLE
Use asyncify to remove callbacks on synchronous functions

### PR DESCRIPTION
**Pull request by clark800**

### CODE REVIEW

File: api/lib/constants.js [hunk 1]
```diff
@@ -74,10 +74,31 @@ var AccountSetResponseFlags = {
     value: ripple.Transaction.flags.AccountSet.DisallowXRP}
 };
 
+var OfferCreateFlags = {
```
Comment (alandotcom) [inline api/lib/constants.js:77]: Should this be part of this commit?

File: api/lib/tx-to-rest-converter.js [hunk 2]
```diff
@@ -3,57 +3,49 @@
...
 
-function TxToRestConverter() {}
-
 // This is just to support the legacy naming of "counterparty", this
 // function should be removed when "issuer" is eliminated
-function renameCounterpartyToIssuer(orderChanges) {
+function renameCounterpartyToIssuerInOrderChanges(orderChanges) {
   return _.mapValues(orderChanges, function(changes) {
     return _.map(changes, function(change) {
+      return utils.renameCounterpartyToIssuerInOrder(change);
```
Comment (alandotcom) [inline api/lib/tx-to-rest-converter.js:17]: I think you included more refactorings than the commit title suggests, maybe?

Comment (clark800) [inline api/lib/tx-to-rest-converter.js:17]: Yes, I always try to throw in some bonuses

File: api/lib/utils.js [hunk 3]
```diff
@@ -122,34 +140,30 @@ function attachDate(api, baseTransactions, callback) {
...
+    return _.map(transactions, function(tx) {
+      return _.assign(tx, {date: data.ledger.close_time});
+    });
+  }
 
-        _.each(_groupedTx[ledger], function(tx) {
-          tx.date = data.ledger.close_time;
-        });
+  function getLedger(ledgerIndex, _callback) {
+    api.remote.requestLedger({ledger_index: ledgerIndex}, _callback);
```
Comment (alandotcom) [inline api/lib/utils.js:150]: Where is `api` defined?

Comment (alandotcom) [inline api/lib/utils.js:150]: Nevermind, didn't see the nesting

Comment (clark800) [inline api/lib/utils.js:150]: Parameter to the outer function

File: api/lib/tx-to-rest-converter.js (hunk 2 above)
Comment (alandotcom) [inline api/lib/tx-to-rest-converter.js:17]: Maybe the commit message can reflect that?

Comment (clark800) [inline api/lib/tx-to-rest-converter.js:17]: Then almost all of my commits messages would end with "& some miscellaneous refactoring". I don't think it's necessary.

**Comment by alandotcom** [comment]
This looks good to me, but I would appreciate if the commit message mentions updating a dependency and adding a new one (https://github.com/lydell/simple-asyncify) with maybe an example of how it's useful

**Comment by alandotcom** [comment]
LGTM

**Pull request merged**